src/Processors/QueryPlan/Optimizations/filterPushDown.cpp

```diff
@@ -118,6 +130,15 @@ tryAddNewFilterStep(QueryPlan::Node * parent_node, QueryPlan::Nodes & nodes, con
     return 3;
 }
 
+static size_t
+tryAddNewFilterStep(QueryPlan::Node * parent_node, QueryPlan::Nodes & nodes, const Names & allowed_inputs,
+                    bool can_remove_filter = true)
+{
+    if (auto split_filter = splitFilter(parent_node, allowed_inputs, 0))
+        return tryAddNewFilterStep(parent_node, nodes, std::move(split_filter), can_remove_filter, 0);
```

Comment:
There is no need for `move`, `tryAddNewFilterStep()` should just accept `const &` on shared ptr.